src/device.cpp

```diff
@@ -374,6 +374,14 @@ void cvk_device::build_extension_ils_list() {
             MAKE_NAME_VERSION(1, 0, 0, "cl_khr_device_uuid"));
     }
 
+    // Enable cl_khr_fp16 if we have 16-bit storage and shaderFloat16
+    if (is_vulkan_extension_enabled(VK_KHR_16BIT_STORAGE_EXTENSION_NAME) &&
+        is_vulkan_extension_enabled(
+            VK_KHR_SHADER_FLOAT16_INT8_EXTENSION_NAME) &&
+        m_features_float16_int8.shaderFloat16) {
```

Comment:
I think we should check for at least `m_features_16bit_storage.storageBuffer16BitAccess` and probably `m_features_16bit_storage.uniformAndStorageBuffer16BitAccess` and/or  `m_features_16bit_storage.storagePushConstant16` for POD argument buffers and POD arguments in push constants.

Reply:
I've added a check for `storageBuffer16BitAccess`.

I don't think checking for uniforms and push constants is necessary. clvk already tells Clspv whether or not these things are supported, and Clspv should automatically fallback to storage buffers for things like POD arguments if necessary.

Reply:
Yes, with Alan's recent-ish changes, uniforms and push constants will be selected only when supported. I agree that checking for storage buffer support is enough.